src/PostgREST/Query/SqlFragment.hs

```diff
@@ -501,7 +501,7 @@ returningF qi returnings =
 
 limitOffsetF :: NonnegRange -> SQL.Snippet
 limitOffsetF range =
-  if range == allRange then mempty else "LIMIT " <> limit <> " OFFSET " <> offset
+  if range == allRange then mempty else " LIMIT " <> limit <> " OFFSET " <> offset
```

Comment:
Oof, this is too subtle. How about using [unwords](https://hackage.haskell.org/package/base-4.21.0.0/docs/Prelude.html#v:unwords) instead?

We have that on https://github.com/PostgREST/postgrest/blob/0aa397fa1d5ded01633fe123efe17b3bdd19f63e/src/PostgREST/Query/SqlFragment.hs#L356-L358

Reply:
The unwords would need to be used at the QueryBuilder.hs level, since the separation is between `orderF` and `limitOffsetF` (both are SQL.Snippet)... Hmm... I can add the separation there to be more explicit, it was like that before:

```haskell
-- QueryBuilder.hs
  groupF qi select relSelect <> " " <>
  orderF qi order <> " " <>
  limitOffsetF readRange
  where
```

Reply:
Or do you mean having something like:

```suggestion
  if range == allRange then mempty else " " <> BS.unwords ["LIMIT", limit, "OFFSET", offset]
```

Reply:
Hmm... still not possible cause `limit` and `offset` are Snippet types. So, I think the only alternative to make the space more explicit would be this one https://github.com/PostgREST/postgrest/pull/4114/files#r2116852368.

Reply:
If you convert `" LIMIT "`and `" OFFSET " ` to `Snippet` with `SQL.sql` , won't `unwords` work?

I'm fine with the other alternative otherwise, it's just too dangerous to leave spaces like that.